### DISCUSSION TITLE
`dRHOds` units used in the eos calculation

### DISCUSSION THREAD

**Discussion by andersy005** [NCAR/pop-tools#143] [Q&A]
From an offline email conversation with @islasimpson

> Can someone advise me what the units are of dRHOds in the eos calculation.  The code is setting it to kg/m^2/degC but I suspect this is incorrect as it seems like instead of degC it should be per some form of salinity unit.  I assumed it would be in PSU's but I'm having to divide by 1000 to get it to make sense.  It's also possible that it should be in degC and I don't really understand what I'm doing which is why I'm raising this in discussion rather than an issue.

https://github.com/NCAR/pop-tools/blob/f8dc20b9962e0fdc7b9a1995503e368ff326e3e7/pop_tools/eos.py#L104-L110

Ccing @matt-long,@mnlevy1981, @dcherian, @klindsay28, @ALDepp who know more about POP than I do :)

**Comment by matt-long** [comment]
@andersy005, you are correct. `S` is salinity, so degC is incorrect. Correct alternatives to consider:
- g/kg
- PSU

I think I would go with PSU.

**Reply by islasimpson** [reply]
Thanks.  I realize now why I couldn't comment yesterday - I hadn't put a category.  Anyway, makes sense it should be in some salinity units.  Are you sure it's in PSU?  When I calculate the anomalous density component associated with salinity using dRHOdS*SALT where SALT is in g/kg I had to divide it by 1000 for the salt and temperature components to add up to the total.  Is it possible that the rate of change with salinity is in kg/m3/kg/kg instead of kg/m3/g/kg?  Or maybe I've just done something weird...

**Comment by mnlevy1981** [comment]
@islasimpson There is a factor of 1000 in the computation of `dRHOdS` in `_compute_eos_coeffs()`:

https://github.com/NCAR/pop-tools/blob/f8dc20b9962e0fdc7b9a1995503e368ff326e3e7/pop_tools/eos.py#L326 

I always bungle things when I try to write out salinity units, and the added complication of having the salinity in the denominator would only exacerbate the problem... but it makes sense to me that `(WORK3 - WORK1 * DENOMK * WORK4) * DENOMK` would be `kg/m3/(g/kg)` and then multiplying by `1000 g / kg` converts it to `(kg/m3)/(kg/kg)`

**Reply by mnlevy1981** [reply]
I should add that I tried to figure out the units of `DENOMK`, but even after comparing the coefficients defined in

https://github.com/NCAR/pop-tools/blob/f8dc20b9962e0fdc7b9a1995503e368ff326e3e7/pop_tools/eos.py#L242-L255

to what's in [MJWR (2003)](https://journals.ametsoc.org/view/journals/atot/20/5/1520-0426_2003_20_730_aaceaf_2_0_co_2.xml) I couldn't figure it out. In section 2, the paper says

> Note that while salinity is a dimensionless quantity, we use the common terminology of 35 psu (standing for practical salinity units) to describe a sa- linity of 0.035 because it tells the reader exactly how the salinity has been calculated

But that could mean that the units of `S` are in `psu` or that the units of `S` are in `kg/kg` but they write things like "Using a million random points drawn from the (S– T–p) cube (0 psu, 40 psu) x (-3 C, 40 C) x (0 db, 104 db)" to make it easier for the reader to understand the ranges

**Reply by islasimpson** [reply]
Thanks for looking into it.  This is not something I know about at all I'm afraid.  All I know is that I had to divide the SALT part by 1000 to get the SALT and TEMP anomalies to add up to the total, which suggests to me it's in kg/m3 per kg/kg.  But since this is all new to me, it's possible I am doing something else wrong.
